### PR TITLE
Add destroy dependent for tasks and service offerings in source

### PR DESCRIPTION
**Pull request by hsong-rh**
https://issues.redhat.com/browse/SSP-2138

### CODE REVIEW

File: app/models/source.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ class Source < ApplicationRecord
   # Service Catalog Inventory Objects
   has_many :service_credential_types
   has_many :service_credentials
-  has_many :service_offerings
+  has_many :service_offerings, :dependent => :destroy
```
Comment (mkanoor) [inline app/models/source.rb:12]: @hsong-rh Will this just mark the service_offering with archived_at and not delete it?

File: app/models/source.rb [hunk 2]
```diff
@@ -25,7 +25,7 @@ class Source < ApplicationRecord
   has_many :service_plans
 
   # Tasks
-  has_many :tasks
+  has_many :tasks, :dependent => :destroy
```
Comment (mkanoor) [inline app/models/source.rb:28]: @hsong-rh This looks ok

File: app/models/source.rb (hunk 1 above)
Comment (hsong-rh) [reply app/models/source.rb:12]: @mkanoor I tested, it's deleted, not archived

File: app/models/source.rb (hunk 2 above)
Comment (hsong-rh) [reply app/models/source.rb:28]: @mkanoor Added more dependent destroys

**Review by mkanoor — APPROVED** [review]

**Pull request merged**